mountpoint-s3/src/fuse.rs

```diff
@@ -25,23 +25,38 @@ pub mod session;
 macro_rules! fuse_error {
     ($name:literal, $reply:expr, $err:expr) => {{
         let err = $err;
-        ::tracing::warn!("{} failed: {:#}", $name, err);
+        match err.level {
+            ::tracing::Level::ERROR => ::tracing::error!("{} failed: {:#}", $name, err),
+            ::tracing::Level::WARN => ::tracing::warn!("{} failed: {:#}", $name, err),
+            ::tracing::Level::INFO => ::tracing::info!("{} failed: {:#}", $name, err),
+            ::tracing::Level::DEBUG => ::tracing::debug!("{} failed: {:#}", $name, err),
+            ::tracing::Level::TRACE => ::tracing::trace!("{} failed: {:#}", $name, err),
+        }
         ::metrics::counter!("fuse.op_failures", 1, "op" => $name);
         $reply.error(err.to_errno());
     }};
 }
```

Comment:
I think this is the same idea as we already have here:

https://github.com/awslabs/mountpoint-s3/blob/e52d68ede985ecdb31f68e8db2beef29b528c8b1/mountpoint-s3-client/src/s3_crt_client.rs#L68-L78

The problem is that while tracing's `event!` lets you provide a level, it needs to be known at compile time. It's a bit annoying.

That said, we should totally factor this out into an `event!` macro, especially since you repeat the same idea twice in this file now (once for `fuse_error` and once for `fuse_unsupported`).

Reply:
Ah, my bad. That's annoying!

OK, then this looks good. Let's factor as an `event!` macro with a short rustdoc explaining why its needed.